lib/uri/query_params/mixin.rb

```diff
@@ -29,9 +29,9 @@ def self.included(base)
           # @since 0.5.2
           #
           def query
-            if @query_params then URI::QueryParams.dump(@query_params)
-            else                  raw_query
-            end
```

Comment:
Could we keep the `if / else` block? It nicely shows that the code falls back to the original implementation if there are no query params set.

Reply:
Sure, your domain, your style rules 😄 

I usually follow the pattern of validating and cutting execution on top, followed by the "default" work that the method is supposed to be doing.